service/src/ConstConfigIOGroup.cpp

```diff
@@ -0,0 +1,424 @@
+        if (it == m_signal_available.end()) {
+            throw Exception("ConstConfigIOGroup::agg_function(): unknown how "
+                            "to aggregate \"" + signal_name + "\"",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        return it->second.agg_function;
+    }
+
+    std::function<std::string(double)>
+    ConstConfigIOGroup::format_function(const std::string &signal_name) const
+    {
+        if (!is_valid_signal(signal_name)) {
+            throw Exception("ConstConfigIOGroup::format_function(): unknown "
+                            "how to format \"" + signal_name + "\"",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        return string_format_double;
+    }
+
+    std::string ConstConfigIOGroup::signal_description(
+        const std::string &signal_name) const
+    {
+        auto it = m_signal_available.find(signal_name);
+        if (it == m_signal_available.end()) {
+            throw Exception("ConstConfigIOGroup::signal_description(): "
+                            "signal_name " + signal_name +
+                            " not valid for ConstConfigIOGroup",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        std::string result;
+        result =  "    description: " + it->second.description + '\n'; // Includes alias_for if applicable
+        result += "    units: " + IOGroup::units_to_string(it->second.units) + '\n';
+        result += "    aggregation: " + Agg::function_to_name(it->second.agg_function) + '\n';
+        result += "    domain: " + platform_topo().domain_type_to_name(GEOPM_DOMAIN_BOARD) + '\n';
+        result += "    iogroup: ConstConfigIOGroup";
+        return result;
+    }
+
+    std::string ConstConfigIOGroup::control_description(
+        const std::string &control_name) const
+    {
+        throw Exception("ConstConfigIOGroup::control_description: there are "
+                        "no controls supported by the ConstConfigIOGroup",
+                        GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+    }
+
+    int ConstConfigIOGroup::signal_behavior(const std::string &signal_name) const
+    {
+        if (!is_valid_signal(signal_name)) {
+            throw Exception("ConstConfigIOGroup::signal_behavior(): "
+                            "signal_name " + signal_name +
+                            " not valid for ConstConfigIOGroup",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        return IOGroup::M_SIGNAL_BEHAVIOR_CONSTANT;
+    }
+
+    std::string ConstConfigIOGroup::name(void) const
+    {
+        return plugin_name();
+    }
+
+    std::string ConstConfigIOGroup::plugin_name(void)
+    {
+        return M_PLUGIN_NAME;
+    }
+
+    std::unique_ptr<IOGroup> ConstConfigIOGroup::make_plugin(void)
+    {
+        return geopm::make_unique<ConstConfigIOGroup>();
+    }
```

Comment:
How will this change for non-empty const configs? Maybe managed similar to other geopm config files?

Reply:
I think I'll be able to get away with still invoking the default constructor once I update it to load the config from a file (which could be provided via an environment variable according to the feature description).